### DISCUSSION TITLE
I need to force install dependency using vercel.json for node.js app

### DISCUSSION THREAD

**Discussion by RS156** [vercel/community#2989] [Help]
### Summary

I have to deploy a node.js application and I need to force install the dependency. Using the Build settings from ui I can set the command  to npm install --force and it worked. however, to run the @vercel/node package i need to use vercel.json. From here, it overrides the install command. Using "installCommand": "npm ci --force", also did not solve my proble. vercel still tries to install without forcing.

If this does not work then i need to fork that pacakage and update the dependcy manually. But I think I may be missing something. ANy help will be appreciated.

Here is my vercel.json for node.js applcation

{
    "version": 2,
    "buildCommand": "echo hello",
    "installCommand": "npm ci --force",
    "builds": [
        {
            "src": "index.js",
            "use": "@vercel/node"
        },
        {
            "src": "dist/**",
            "use": "@vercel/static"
        }
    ],
    
    "routes": [
      
        {
            "src": "/",
            "dest": "dist/index.html"
        },
        {
            "src": "/(.+)",
            "dest": "index.js"
        }
    ]
}

### Steps to Reproduce

1. Set up a Express.js project with some api.
2. setup vercel.json as follows 
{
    "version": 2,
    "buildCommand": "echo hello",
    "installCommand": "npm ci --force",
    "builds": [
        {
            "src": "index.js",
            "use": "@vercel/node"
        },
        {
            "src": "dist/**",
            "use": "@vercel/static"
        }
    ],
    
    "routes": [
      
        {
            "src": "/",
            "dest": "dist/index.html"
        },
        {
            "src": "/(.+)",
            "dest": "index.js"
        }
    ]
}

3. Try to run vercel --prod

expected - Vercel should instal dependencies with npm install --force command
Actual - Vercel is not installing dependencies forcefully. If we used Build Settings from UI, then dependencies are installed forcefully. But, we cannot setup node.js using UI.

**Comment by amyegan** [comment]
Hi @RS156. You can set `npm install --force` as the `installCommand` in vercel.json if you prefer to set it there. Or you can remove the "buildCommand" and "installCommand" from vercel.json and set them in the UI instead.

**Reply by RS156** [reply]
I tried it first. This is my vercel.json file.

{
    "version": 2,
    "buildCommand": "echo hello",
    "installCommand": "npm ci --force",
    "builds": [
        {
            "src": "index.js",
            "use": "@vercel/node"
        },
        {
            "src": "dist/**",
            "use": "@vercel/static"
        }
    ],
    
    "routes": [
      
        {
            "src": "/api/(.*)",
            "dest": "index.js"
        },
        {
            "src": "/",
            "dest": "dist/index.html"
        },
        {
            "src": "/(.+)",
            "dest": "dist/$1"
        }
    ]
}

I tried npm install --force as well on vercel.json, still the dependencies were downloaded by default command. In UI if i used --force flag then dependencies were downloaded forcefully. It may be an issue with vercel.json configuration.
